backend/prompt_studio/prompt_studio_registry_v2/prompt_studio_registry_helper.py

```
@@ -143,6 +143,31 @@ def get_tool_by_prompt_registry_id(
+        notably ``challenge_llm``, which falls back to the default profile's LLM
+        when the project set none - and stores them under
+        ``tool_metadata[tool_settings]``. ``Tool`` (built from ``tool_spec`` /
+        ``tool_property``) does not carry them, so callers that only have a
+        ``Tool`` cannot see the resolved values.
+
+        Returns an empty dict when the registry row is missing or carries no
+        settings, so callers can treat "no resolved settings" as a no-op.
+        """
+        try:
+            prompt_registry_tool = PromptStudioRegistry.objects.get(pk=prompt_registry_id)
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)